drivers/clock_control/clock_control_it8xxx2.c

```diff
@@ -0,0 +1,442 @@
+
+	volatile uint8_t _pll_ctrl __unused;
+
+	ecpm_regs->PLLCTRL = mode;
+	/*
+	 * for deep doze / sleep mode
+	 * This load operation will ensure PLL setting is taken into
+	 * control register before wait for interrupt instruction.
+	 */
+	_pll_ctrl = ecpm_regs->PLLCTRL;
+}
+
+void __soc_ram_code chip_run_pll_sequence(const struct pll_config_t *pll)
+{
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	/* Enable HW timer to wakeup chip from the sleep mode */
+	timer_5ms_one_shot();
+	/*
+	 * Configure PLL clock dividers.
+	 * Writing data to these registers doesn't change the
+	 * PLL frequency immediately until the status is changed
+	 * into wakeup from the sleep mode.
+	 * The following code is intended to make the system
+	 * enter sleep mode, and wait HW timer to wakeup chip to
+	 * complete PLL update.
+	 */
+	ecpm_regs->PLLFREQR = pll->pll_setting;
+	/* Pre-set FND clock frequency = PLL / 3 */
+	ecpm_regs->SCDCR0 = (2 << 4);
+	/* JTAG and EC */
+	ecpm_regs->SCDCR3 = (pll->div_jtag << 4) | pll->div_ec;
+	/* Chip sleep after wait for interrupt (wfi) instruction */
+	chip_pll_ctrl(CHIP_PLL_SLEEP);
+	/* Chip sleep and wait timer wake it up */
+	__asm__ volatile  ("wfi");
+	/* New FND clock frequency */
+	ecpm_regs->SCDCR0 = pll->div_fnd << 4;
+	/* Chip doze after wfi instruction */
+	chip_pll_ctrl(CHIP_PLL_DOZE);
+	/* UART */
+	ecpm_regs->SCDCR1 = pll->div_uart;
+	/* SSPI and SMB */
+	ecpm_regs->SCDCR2 = (pll->div_sspi << 4) | pll->div_smb;
+	/* USBPD and PWM */
+	ecpm_regs->SCDCR4 = (pll->div_usbpd << 4) | pll->div_pwm;
+}
+
+static void chip_configure_pll(const struct pll_config_t *pll)
+{
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	/* Re-configure PLL clock or not. */
+	if (((ecpm_regs->PLLFREQR & PLLFREQ_MASK) != pll->pll_setting) ||
+	    ((ecpm_regs->SCDCR0 & FNDFREQ_MASK) != (pll->div_fnd << 4)) ||
+	    ((ecpm_regs->SCDCR3 & EC_FREQ_MASK) != pll->div_ec)) {
+#ifdef CONFIG_ESPI
+		/*
+		 * We have to disable eSPI pad before changing
+		 * PLL sequence or sequence will fail if CS# pin is low.
+		 */
+		espi_it8xxx2_enable_pad_ctrl(ESPI_IT8XXX2_SOC_DEV, false);
+#endif
+		/* Run change PLL sequence */
+		chip_run_pll_sequence(pll);
+#ifdef CONFIG_ESPI
+		/* Enable eSPI pad after changing PLL sequence */
+		espi_it8xxx2_enable_pad_ctrl(ESPI_IT8XXX2_SOC_DEV, true);
+#endif
+	}
+}
+
+#endif /* CONFIG_SOC_IT8XXX2_PLL_FLASH_48M */
+
+static int it8xxx2_clock_control_init(const struct device *dev)
+{
+	clock_config = dev->config;
+	int pllfreq = clock_config->pll_freq;
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	/* SWUC Clock Gating */
+	ecpm_regs->CGCTRL2R |= SWUC_CLOCK_GATING;
+	/* PECI, SSPI, DBGR Clock Gating */
+	ecpm_regs->CGCTRL3R |= PECI_SSPI_DBGR_CLOCK_GATING;
+	/* SMB/I2C Clock Gating */
+	ecpm_regs->CGCTRL4R |= SMB_ALL_CHANNELS_CLOCK_GATING;
+
+
+	if (!(DT_NODE_HAS_STATUS(DT_NODELABEL(usbpd0), okay))) {
```

Comment:
`!okay` -> `disabled`? Would simplify this line a little bit.

Reply:
You're absolutely right, sir. I'll refine it ASAP.